server/mailet/dkim/src/main/java/org/apache/james/jdkim/mailets/DKIMSign.java

```diff
@@ -153,38 +149,23 @@ public void init() throws MessagingException {
 
     public void service(Mail mail) throws MessagingException {
         DKIMSigner signer = new DKIMSigner(getSignatureTemplate(), getPrivateKey());
-        SignatureRecord signRecord = signer
-                .newSignatureRecordTemplate(getSignatureTemplate());
-        try {
-            BodyHasher bhj = signer.newBodyHasher(signRecord);
-            MimeMessage message = mail.getMessage();
-            Headers headers = new MimeMessageHeaders(message);
-            try {
-                OutputStream os = new HeaderSkippingOutputStream(bhj.getOutputStream());
-                if (forceCRLF) {
-                    os = new CRLFOutputStream(os);
-                }
-                try (MimeMessageInputStream stream = new MimeMessageInputStream(message)) {
-                    stream.transferTo(os);
-                }
-            } catch (IOException e) {
-                throw new MessagingException("Exception calculating bodyhash: " + e.getMessage(), e);
-            } finally {
-                try {
-                    bhj.getOutputStream().close();
-                } catch (IOException e) {
-                    throw new MessagingException("Exception calculating bodyhash: " + e.getMessage(), e);
-                }
-            }
-            String signatureHeader = signer.sign(headers, bhj);
+        MimeMessage message = mail.getMessage();
+
+        try (MimeMessageInputStream stream = new MimeMessageInputStream(message)) {
+            String signatureHeader = forceCRLF ?
+                    signer.sign(new CRLFInputStream(stream)) :
```

Comment:
I'm having a doubt here : the original implementation was using a `HeaderSkippingOutputStream` whereas I apply the CRLF normalization to the whole stream and I'm not sure its valid.

Reply:
the same applied for the verifier implementation